### ISSUE TITLE
BUG: `na_position` ignored when sorting `MultiIndex` with `level!=None`

### ISSUE THREAD

**Issue by gszep** [pandas-dev/pandas#51612]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [x] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
df = pd.DataFrame([1,2],index=pd.MultiIndex.from_tuples([(1,1),(1,pd.NA)]))

df.sort_index()
#        0
# 1 1    1
#   NaN  2

df.sort_index(level=[0,1])
#        0
# 1 NaN  2
#   1    1

# na_position seems to be ignored
df.sort_index(level=[0,1],na_position="last")
#        0
# 1 NaN  2
#   1    1
```

### Issue Description

Sorting should be consistent whether or not all levels in `level` kwarg are provided. It appears that `na_position` is ignored when providing `level`.

### Expected Behavior

`na_position="last"` should be the default behaviour no matter whether `level` is provided or not

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : b69bd0755d654c0dba42042df7b2251918274327
python           : 3.10.4.final.0
python-bits      : 64
OS               : Linux
OS-release       : 6.0.12-76060006-generic
Version          : #202212290932~1674139725~22.04~ca93ccf SMP PREEMPT_DYNAMIC Thu J
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.1.0.dev0+47.gb69bd0755d
numpy            : 1.23.5
pytz             : 2022.7.1
dateutil         : 2.8.2
setuptools       : 65.6.3
pip              : 23.0
Cython           : None
pytest           : 7.2.0
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.8.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2022.11.0
gcsfs            : None
matplotlib       : 3.6.2
numba            : 0.56.4
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.11.0
scipy            : 1.9.3
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : None
qtpy             : None
pyqt5            : None
</details>

**Comment by phofl** [comment]
Yeah, can you rephrase as enhancement request?

**Comment by AlexKirko** [comment]
At the very least, it would be nice to give the user a warning in this case, as there is little reason for them to expect the argument being ignored.

**Comment by phofl** [comment]
I'd prefer an implementation. Alternatively I'd rather raise than warn
